cms/server/AdminWebServer.py

```diff
@@ -759,17 +1087,20 @@ def post(self, task_id):
         if try_commit(self.sql_session, self):
             self.redirect("/task/%s" % task_id)
         else:
-            self.redirect("/add_statement/%s" % task_id)
+            self.redirect(fallback_page)
 
 
 class DeleteStatementHandler(BaseHandler):
     """Delete a statement.
 
     """
-    def get(self, statement_id):
+    def get(self, task_id, statement_id):
         statement = self.safe_get_item(Statement, statement_id)
-        task = statement.task
-        self.contest = task.contest
+        task = self.safe_get_item(Task, task_id)
+
+        # Additional check.
+        if task is not statement.task:
+            raise tornado.web.HTTPError(404)
```

Comment:
Uhm if I recall correctly that check was more like an assert. The only case I can think of that happening is when the user visits an URL that "purposely" mixes up a statement_id and a task_id that are not related (and that doesn't happen when "browsing" the admin panel), so I thought that a 404 response was the most semantically accurate thing to do.